### PR TITLE
Make Header global

### PR DESCRIPTION
**Pull request by nachoiacovino**
Based on [this comment](https://github.com/clerkinc/clerk-nextjs-starter/pull/7#pullrequestreview-624513217).

I reused a Header component from Clerk_UI which I thought it was exactly what we needed, I first added it to the UserProfile page but then I realized it made more sense to make that Header global, so I moved the Header component to _app.js.

This is the new look:

/
image

/user
image

Please tell me if there's a need for any change.

### CODE REVIEW

**Comment by colinclerk** [comment]
Looking great - definitely the right decision to pull Header into a component.  I'm adding @nikosdouvlis here for help reviewing.  My questions:
 * Should header styles be pulled into its own css module?
 * Perhaps omit the Clerk logo and replace it with the text, "Your Application"
 * I'm not sure the Header should be on the /sign-in and /sign-up pages.  Maybe pull the component into a separate file, then mount it manually on the /index page and /user page?  Or, mount it manually on the /index page, but leave it with <SignedIn> on _app so it's automatically added to /user (and any other pages the developer creates).

**Comment by nikosdouvlis** [comment]
Thanks @colinclerk, looking great indeed 🎉  I've only got a couple of minor points - we've discussed a bit with @nachoiacovino already, but we'll go through them together on a quick call tomorrow morning!

File: components/Layout.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import Header from "./Header";
+
+const Layout = ({ children }) => (
+  <>
+    <Header />
```
Comment (nikosdouvlis) [inline components/Layout.js:5]: Good choice to make the header global! Having a way go back to the home page while on the login page makes a lot of sense

File: styles/Header.module.css [hunk 2]
```diff
@@ -0,0 +1,31 @@
+.header {
```
Comment (nikosdouvlis) [inline styles/Header.module.css:1]: My only suggestion is that we move the css modules to same directory as their components. Thats the convention we and most OS project use, so lets go with that! Of course, this can be part of a different PR.

**Review by nikosdouvlis — APPROVED** [review]
Well done! 💯

**Review by colinclerk — APPROVED** [review]

File: styles/Header.module.css (hunk 2 above)
Comment (nachoiacovino) [reply styles/Header.module.css:1]: I'll open a different PR for it next week! 

Thank for your suggestions Nikos, they were really helpful.
